dot/sync/chain_sync.go

```diff
@@ -932,6 +936,34 @@ func (cs *chainSync) validateJustification(bd *types.BlockData) error {
 	return nil
 }
 
+func (cs *chainSync) getHighestBlock() (int64, error) {
+	cs.RLock()
+	ps := cs.peerState
+	cs.RUnlock()
+
+	if len(ps) == 0 {
+		return 0, errNoPeers
+	}
+
+	highestBlock := big.NewInt(-1)
+
+	for _, ps := range ps {
+		if ps.number == nil {
+			continue
+		}
+		if ps.number.Cmp(highestBlock) < 0 { // ps.number < highestBlock: https://pkg.go.dev/math/big#Int.Cmp
+			continue
+		}
```

Comment:
```suggestion
		if ps.number == nil || ps.number.Cmp(highestBlock) < 0 {
			continue
		}
```

Reply:
Yeah, I did some tests here and I thought that I could get a panic when the `ps.number` is `nil` but I didn't. Good catch!

Done in 5afd977